src/audits/performance/user-timings.js

```diff
@@ -48,36 +48,71 @@ class UserTimings extends Audit {
       });
     }
 
-    let timingsCount = 0;
+    let userTimings = [];
+    let navigationStartTime;
+    let measuresStartTimes = {};
+    let traceStart = false;
 
     // Fetch blink.user_timing events from the tracing data
     const timelineModel = new TimelineModel(artifacts.traceContents);
     const modeledTraceData = timelineModel.timelineModel();
-    const key = [...modeledTraceData.mainThreadAsyncEvents().keys()].find(
-      key => key.title === 'User Timing'
-    );
-    let userTimings = modeledTraceData.mainThreadAsyncEvents().get(key);
 
-    // Reduce events to record only useful information
-    if (typeof userTimings !== 'undefined') {
-      timingsCount = userTimings.length;
-      userTimings = userTimings.map(ut => {
-        return {
-          name: ut.name,
-          startTime: ut.startTime,
-          endTime: ut.endTime,
-          duration: ut.duration.toFixed(2) + 'ms',
-          args: ut.args
-        };
-      });
-    }
+    // Get all blink.user_timing events
+    // The event phases we are interested in are mark and instant events (R, i, I)
+    // and duration events which correspond to measures (B, b, E, e).
+    // @see https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#
+    modeledTraceData.mainThreadEvents()
+    .filter(ut => ut.hasCategory('blink.user_timing') || ut.name === 'TracingStartedInPage')
```

Comment:
how big is the filtered array from this step? If it's rather large it may not worth be worth creating it and you could just add the category/name check inside the `forEach` loop